awx_collection/test/awx/test_team.py

```diff
@@ -47,7 +48,10 @@ def test_modify_team(run_module, admin_user):
     }, admin_user)
     team.refresh_from_db()
     result.pop('invocation')
+    result.pop('field_changes')
     assert result == {
+        "needed_update": True,
+        "updated": False,
```

Comment:
This isn't something consistent enough that we could document in the `RETURNS` string. If it won't buy us that, I don't see why we should do it.

If it's created, then it returns the `created` key, and if it's updated it returns the `updated` key. But neither of these keys are present if it's not created or updated. The proposal in https://github.com/ansible/awx/issues/6367 is that other status like `approved` would also be present. Then should `started` and `finished` be status types too?

I'd advocate that returning `state` might be best, even if it's a repeat of the input parameter. That's documentable, and not always obvious because it has a default.

Reply:
I added some of this as almost "Tracking" so that you could look at the returned data and know what it did. I thought that might also help us prepare for diff mode. i.e. instead of just `changed: True` why was it actually changed. This has also helped in development of Tower credential to know why a credential is being reported as changed.

I think we should be able to set them in the returned field because they can have an optional description on them (I believe). i.e.
```
needed_update:
    description: If the item needed to be updated
    returned: when item already exists
    type: bool
    sample: True
```

Reply:
If there's going to be churn on this, we might be better off to rewrite the (unit) test assertions so that they just don't assert on the full data - and develop something after things have stabilized. For the purposes of testing / tracking specific behaviors, that makes sense to me, I just worry if the key naming seems too general. But I also can't connect the dots from these changes to the JT module conversion. A separate PR might be nice.

Reply:
The only reason they were in here was because I was hitting a condition where the JT was always returning changed = True. Instead of doing my old: `with open('/tmp/john', 'w') as f ....` I tried to get smart and add these additional fields into the module_util so that I could tell what it was trying to do internally. Incidentally, I needed these things subsequently for tower_credential to track down the issue with the inputs hash.